### ISSUE TITLE
amp-lightbox-gallery or auto-lightboxed images should be keyboard accessible

### ISSUE THREAD

**Issue by cathyxz** [ampproject/amphtml#25317]
We should consider doing this by adding a tabindex and/or appropriate role attribute, but first make sure that panning and zooming in lightbox has a good keyboard experience.

**Comment by westonruter** [comment]
I just came across this myself.

I noticed `amp-img` had `on="tap:…"` attribute when looking in DevTools but I was surprised to see no `tabindex=0` or `role=button`, and yet there was no AMP validation error. Then I saw that the `on` attribute is added by the component JS, so that's why it's not flagged by the validator.

Example to reproduce this issue:

```html
<!doctype html>
<html ⚡>
<head>
  <meta charset="utf-8">
  <title>My AMP Page</title>
  <link rel="canonical" href="self.html" />
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-lightbox-gallery" src="https://cdn.ampproject.org/v0/amp-lightbox-gallery-0.1.js"></script>
</head>
<body>  	
  	<span tabindex=0>Focusable element before amp-img</span>
	<amp-img src="https://blog.amp.dev/wp-content/uploads/2020/03/AMP_camp_Blog.png" height="1" width="2" layout="responsive" lightbox=""></amp-img>
  	<span tabindex=0>Focusable element after amp-img</span>
</body>
</html>
```

If I manually add `tabindex=0` and `role=button` to the `amp-img` then it does allow the `amp-img` lightbox to be opened by hitting enter or spacebar, but with one additional oddity: hitting the spacebar causes the lightbox to re-open when I would have expected instead for the lightbox to close (or do nothing).
